pytorch_lightning/trainer/supporters.py

```diff
@@ -403,6 +404,10 @@ def create_loader_iters(dataloader: DataLoader, state_dict: Dict) -> Iterator:
             if isinstance(dataloader, CycleIterator):
                 dataloader = dataloader_to_iter_on.loader
 
+            # dataset states are collected across all ranks
+            rank = torch.distributed.get_rank() if distributed_available() else 0
```

Comment:
alternatively, we could access Lightnings global rank variable, i.e., `_get_rank()` to make it more strategy agnostic, although granted it is not much cleaner.

Reply:
Yes, I believe we should do this when we have a cleaner API.

Right now, I still want the code to be unittested on this own and we haven't tested fault tolerant on TPU, IPU, horovod yet, so I believe there might be more caviats.